### PR TITLE
delegate gets informed when the empty view becomes hidden/visible

### PR DESCRIPTION
**Pull request by 343max**

### CODE REVIEW

File: UITableView+NXEmptyView/NXEmptyTableViewDelegate.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+//  Copyright (c) 2013 nxtbgthng.com. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@class UITableView;
+
+@protocol NXEmptyTableViewDelegate <NSObject>
+
+- (void)tableView:(UITableView *)tableView emptyViewWillBecomeHidden:(BOOL)hidden;
```
Comment (toto) [inline UITableView+NXEmptyView/NXEmptyTableViewDelegate.h:15]: If this is called on hide and unhide a better name would be:

`- (void)tableView:(UITableView *)tableView emptyViewVisibilityWillChange:(BOOL)hidden;`

also maybe adding a did method makes sense as well.

`- (void)tableView:(UITableView *)tableView emptyViewVisibilityDidChange:(BOOL)hidden;`

File: UITableView+NXEmptyView/UITableView+NXEmptyView.m [hunk 2]
```diff
@@ -127,7 +139,12 @@ - (void)nxEV_updateEmptyView;
     }
     
     // show / hide empty view
-    emptyView.hidden = !emptyViewShouldBeShown;
+    
+    if (emptyView.hidden != !emptyViewShouldBeShown &&
+        [self.nxEV_delegate respondsToSelector:@selector(tableView:emptyViewWillBecomeHidden:)]) {
+        [self.nxEV_delegate tableView:self emptyViewWillBecomeHidden:!emptyViewShouldBeShown];
+        emptyView.hidden = !emptyViewShouldBeShown;
```
Comment (stigi) [inline UITableView+NXEmptyView/UITableView+NXEmptyView.m:146]: that's a bug. with this the empty view will only change visibility when the delegate is set & implements the selector.

File: UITableView-NXEmptyView.podspec [hunk 3]
```diff
@@ -1,14 +1,14 @@
 Pod::Spec.new do |s|
   s.name     = 'UITableView-NXEmptyView'
   s.platform = :ios
-  s.version  = '0.1.4'
+  s.version  = '0.1.5'
```
Comment (stigi) [inline UITableView-NXEmptyView.podspec:4]: changing the pods version number is not really semantically connected to the changes in this pull request.  
pull requests should add features and the version number should be bumped on release (that's after the features were merged).

Comment (stigi) [inline UITableView-NXEmptyView.podspec:4]: but not a big issue, just let it in.

**Comment by stigi** [comment]
would it maybe be smart to extend the UITableViewDelegate protocol for this?

**Comment by calimarkus** [comment]
Hab auch erstmal ans UITableViewDelegate gedacht. Dann kann man die beiden Delegates (tableview, emptyview) aber nicht an verschiedenen Stellen nutzen. Is die Frage ob man das braucht.

Andere Idee: Wie wärs wenn man einfach KVO auf hidden nutzt? Dann spart man sich das in der Implementierung hier.

**Comment by stigi** [comment]
@jaydee3 :+1:

**Pull request closed without merging**